gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,191 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
+   * DnsRecordOptions}. Returns null if the zone cannot be found.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(BigInteger zoneId, DnsRecordListOption... options);
+
+  /**
+   * Retrieves the metadata about the current project. The returned fields can be optionally
+   * restricted by specifying {@code ProjectOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects/get">Cloud DNS Projects: get</a>
+   */
+  ProjectInfo getProjectInfo(ProjectGetOption... fields);
+
+  /**
+   * Returns the current project id.
+   */
+  String getProjectId();
+
+  /**
+   * Returns the current project number.
+   */
+  BigInteger getProjectNumber();
+
+  /**
+   * Submits a change requests for applying to the zone identified by ID to the service. The
+   * returned object contains the following read-only fields supplied by the server: id, start time
+   * and status. time, id, and list of name servers. The returned fields can be modified by {@code
+   * ChangeRequestFieldOptions}. Returns null if the zone is not found.
+   *
+   * @return ChangeRequest object representing the new change request or null if zone is not found
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/create">Cloud DNS Changes: create</a>
+   */
+  ChangeRequest applyChangeRequest(ChangeRequest changeRequest, BigInteger zoneId,
+                                   ChangeRequestOption... options);
+
+  /**
+   * Submits a change requests for applying to the zone identified by name to the service. The
+   * returned object contains the following read-only fields supplied by the server: id, start time
+   * and status. time, id, and list of name servers. The returned fields can be modified by {@code
+   * ChangeRequestFieldOptions}. Returns null if the zone is not found.
+   *
+   * @return ChangeRequest object representing the new change request or null if zone is not found
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/create">Cloud DNS Changes: create</a>
+   */
+  ChangeRequest applyChangeRequest(ChangeRequest changeRequest, String zoneName,
+                                   ChangeRequestOption... options);
+
+  /**
+   * Retrieves updated information about a change request previously submitted for a zone identified
+   * by ID. Returns null if the zone or request cannot be found.
+   *
+   * <p>The fields to be returned using {@code ChangeRequestFieldOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/get">Cloud DNS Chages: get</a>
+   */
+  ChangeRequest getChangeRequest(ChangeRequest changeRequest, BigInteger zoneId,
```

Comment:
Is there a benefit in passing the whole `ChangeRequest` object? otherwise I would prefer passing just name/id